### ISSUE TITLE
[Protocol] Failed to open session: client_id_unavailable

### ISSUE THREAD

**Issue by viest** [emqx/emqx#2580]
#### Environment

- OS: 
- Erlang/OTP: R21/10.3.4
- EMQ:  v3.1.1

#### Description

```yml
emqtx_worker:
    image: emqx/emqx:v3.1.1
    ulimits:
      nofile:
        soft: 60000
        hard: 100000
      nproc:
        soft: 60000
        hard: 100000
    environment:
      - "EMQX_NAME=woker"
      - "EMQX_ALLOW_ANONYMOUS=false"
      - "EMQX_LOADED_PLUGINS=emqx_recon,emqx_retainer,emqx_management,emqx_dashboard,emqx_auth_redis,emqx_web_hook"

      - "EMQX_AUTH__REDIS__SERVER=xxxxx:6379"
      - "EMQX_AUTH__REDIS__AUTH_CMD=xxxxx"

      - "EMQX_WEB__HOOK__API__URL=http://xxxxxxx"

      - "EMQX_ZONE__EXTERNAL__IDLE_TIMEOUT=2s"

      - "EMQX_LISTENER__WS__EXTERNAL__PROXY_PROTOCOL=on"
      - "EMQX_LISTENER__WS__EXTERNAL_MAX_CONNECTIONS=1024000"
      - "EMQX_LISTENER__WS__EXTERNAL__PROXY_PROTOCOL_TIMEOUT=1s"
      - "EMQX_LISTENER__WS__EXTERNAL__PROXY_PORT_HEADER=X-Forwarded-Port"
      - "EMQX_LISTENER__WS__EXTERNAL__PROXY_ADDRESS_HEADER=X-Forwarded-For"

      - "EMQX_LISTENER__TCP__EXTERNAL__PROXY_PROTOCOL=on"
      - "EMQX_LISTENER__TCP__EXTERNAL_MAX_CONNECTIONS=1024000"
      - "EMQX_LISTENER__TCP__EXTERNAL__PROXY_PROTOCOL_TIMEOUT=1s"

      - "EMQX_NODE__COOKIE=xxxxxxxxxxxx"

      - "EMQX_CLUSTER__DISCOVERY=dns"
      - "EMQX_CLUSTER__DNS__NAME=emqx_leader.emqx"
      - "EMQX_CLUSTER__DNS__APP=emqx"
      - "EMQX_CLUSTER__AUTOHEAL=on"
      - "EMQX_CLUSTER__AUTOCLEAN=5m"

      - "EMQX_WAIT_TIME=60"
```

集群启动一段时间（十几个小时之后），所有客户端都无法连接.............

![image](https://user-images.githubusercontent.com/14308107/58530182-5e9dd380-8210-11e9-9a0b-bae45cc8619d.png)

**Comment by mdrobny** [comment]
Is this fixed in 3.2-rc.1 already or not yet?

**Comment by Pilchards** [comment]
@turtleDeng Is this fixed in 3.2-rc.1 ？

**Comment by turtleDeng** [comment]
Can't reproduce the problem, please provide the complex scene and provide a more detailed LOG, thank you!

**Comment by king-wood** [comment]
> Can't reproduce the problem, please provide the complex scene and provide a more detailed LOG, thank you!

maybe try it in k8s.

**Comment by turtleDeng** [comment]
It has released at v3.2.1
